src/Cli/Microsoft.DotNet.Cli.Utils/RuntimeEnvironment.cs

```diff
@@ -138,6 +162,36 @@ private static DistroInfo LoadDistroInfo()
             return result;
         }
 
+        private static DistroInfo LoadDistroInfoFromIllumos()
+        {
+            DistroInfo result = null;
+            // examples:
+            //   on OmniOS
+            //       SunOS 5.11 omnios-r151018-95eaa7e
+            //   on OpenIndiana Hipster:
+            //       SunOS 5.11 illumos-63878f749f
+            //   on SmartOS:
+            //       SunOS 5.11 joyent_20200408T231825Z
+            var versionDescription = RuntimeInformation.OSDescription.Split(' ')[2];
+            switch (versionDescription)
+            {
+                case string version when version.StartsWith("omnios"):
+                    result.Id = "OmniOS";
```

Comment:
@am11 @eerhardt Way late, but won't this be a null reference?  This was caught by a code scan tool we ran.

Reply:
Looks like it to me... @am11 thoughts?

Reply:
Ah it will. the testing was blocked by System.Diagnostics.Process port on this platform. I have created #14395.

Reply:
Ah it will. the testing was blocked by System.Diagnostics.Process port on this platform. I have opened #14395.